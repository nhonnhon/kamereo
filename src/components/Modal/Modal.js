import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './Modal.scss'

class Modal extends Component {
  static propTypes = {
    onClose: PropTypes.func,
    children: PropTypes.node,
    className: PropTypes.string,
    hideCloseBtn: PropTypes.bool,
  }

  state = {
    isActive: this.props.isActive,
  }

  componentWillReceiveProps(nextProps) {
    this.setState(
      {
        isActive: nextProps.isActive,
      },
      () => {
        this.state.isActive
          ? document.querySelector('body').classList.add('modal-open')
          : document.querySelector('body').classList.remove('modal-open')
      }
    )
  }

  close = () => {
    const { onClose } = this.props

    this.setState(
      {
        isActive: false,
      },
      () => {
        document.querySelector('body').classList.remove('modal-open')
        onClose && onClose()
      }
    )
  }

  render() {
    const { className, hideCloseBtn, children } = this.props
    const { isActive } = this.state

    return (
      <div className={`modal ${isActive ? 'modal-open' : ''} ${className ? className : ''}`}>
        <div className="modal-dialog">
          <div className={`modal-content`}>
            {!hideCloseBtn && (
              <div className="cancel-btn-modal" onClick={this.close}>
                <div className="close-button">X</div>
              </div>
            )}
            {children}
          </div>
        </div>
      </div>
    )
  }
}

export default Modal
