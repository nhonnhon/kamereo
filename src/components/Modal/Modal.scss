@import '../../assets/css/base.scss';

.modal {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1050;
  overflow: hidden;
  outline: 0;
  background-color: rgba(0, 0, 0, 0.5);
  opacity: 0;
  transition: opacity 0.5s;
  -webkit-transition: opacity 0.5s;
  display: none;
  &.modal-open {
    overflow-x: hidden;
    overflow-y: auto;
    opacity: 1;
    display: block;
  }
}

.modal-dialog {
  margin: 1.75rem auto;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  width: auto;
  min-height: calc(100% - (0.5rem * 2));
  max-width: calc(100vw - 30px);
  transform: translate(0, -25%);
  -webkit-transform: translate(0, -25%);
  transition: transform 0.5s;
  -webkit-transition: transform 0.5s;
  @media (min-width: 576px) {
    min-height: calc(100% - (1.75rem * 2));
    max-width: 500px;
  }
  @media (min-width: 992px) {
    max-width: 975px;
  }
}

.modal-open {
  .modal-dialog {
    -webkit-transform: translate(0, 0);
    transform: translate(0, 0);
  }
}

.modal-content {
  position: relative;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  pointer-events: auto;
  background-color: #fff;
  background-clip: padding-box;
  border-radius: 0.3rem;
  outline: 0;
  padding: 15px;
}

.cancel-btn-modal {
  display: flex;
  justify-content: flex-end;
}

.close-button {
  padding: 10px 12px;
  border-radius: 50%;
  border: 1px solid $gray1;
  cursor: pointer;
}
