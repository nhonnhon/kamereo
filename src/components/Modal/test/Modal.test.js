import React from 'react'
import { create } from 'react-test-renderer'
import Modal from '../Modal'

describe('Modal component', () => {
  test('Matches the snapshot', () => {
    const modal = create(<Modal />)
    expect(modal.toJSON()).toMatchSnapshot()
  })
})
