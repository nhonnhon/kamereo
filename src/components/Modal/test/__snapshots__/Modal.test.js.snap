// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Matches the snapshot 1`] = `
<div
  className="modal  "
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-content"
    >
      <div
        className="cancel-btn-modal"
        onClick={[Function]}
      >
        <div
          className="close-button"
        >
          X
        </div>
      </div>
    </div>
  </div>
</div>
`;
